src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1078,12 +1078,12 @@ public void UpdateMainChain(Block[] processedBlocks, bool wereProcessed)
                     _headerCache.Set(block.Hash, block.Header);
                 }
 
-                MoveToMain(processedBlocks[i], batch, wereProcessed);
+                MoveToMain(processedBlocks[i], batch, wereProcessed, forceUpdateHeadBlock && i == processedBlocks.Length - 1);
```

Comment:
this could use some comments or a named variable to explain the reasoning behind i == processedBlocks.Length - 1

Reply:
Only force last one as head